### PR TITLE
CI: fix clippy issues

### PR DESCRIPTION
**Pull request by muzarski**
There were two clippy checks that were failing:
- not using `unwrap_or_default` in the code generated by `darling` crate. See the corresponding issue: https://github.com/TedDriggs/darling/issues/293
- usage of legacy numeric constant - `std::i16::MAX`, instead of `i16::MAX`

This PR fixes the issues and makes CI pass.

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- ~[ ] I added relevant tests for new features and bug fixes.~
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- ~[ ] I have provided docstrings for the public items that I want to introduce.~
- ~[ ] I have adjusted the documentation in `./docs/source/`.~
- ~[ ] I added appropriate `Fixes:` annotations to PR description.~

### CODE REVIEW

File: scylla-macros/src/serialize/row.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// See: https://github.com/TedDriggs/darling/issues/293
+#![allow(clippy::manual_unwrap_or_default)]
```
Comment (Lorak-mmk) [inline scylla-macros/src/serialize/row.rs:2]: We need to remove those after the issue is fixed in darling. Please open an issue about it so we don't forget

Comment (Lorak-mmk) [reply scylla-macros/src/serialize/row.rs:2]: Also do we need it on the whole file? Isn't it possible to only have it in problematic places?

**Review by Lorak-mmk — CHANGES_REQUESTED** [review]

**Comment by muzarski** [comment]
It looks like GH actions used the old version of clippy: `clippy 0.1.78 (9b00956 2024-04-29)`.
My version locally is: `clippy 0.1.79 (129f3b9 2024-06-10)`.

That's weird, because the runners used for CI in `cql-stress` already use newest version of clippy.

File: scylla-macros/src/serialize/row.rs (hunk 1 above)
Comment (muzarski) [reply scylla-macros/src/serialize/row.rs:2]: > Also do we need it on the whole file? Isn't it possible to only have it in problematic places?

I tried putting it in the following places:
- next to the fields marked with `#[darling(default)]` attribute (tried both after and before darling attribute)
- next to the struct with marked fields

None of these worked. Do you have any more ideas?

Comment (piodul) [reply scylla-macros/src/serialize/row.rs:2]: You can also try putting the `allow` before or after the `#[derive(FromAttributes)]` part.

Comment (muzarski) [reply scylla-macros/src/serialize/row.rs:2]: > You can also try putting the `allow` before or after the `#[derive(FromAttributes)]` part.

Unfortunately, this does not work

**Comment by Lorak-mmk** [comment]
Afaik rustup is present in runner images. Maybe we should explicitly install latest stable using rustup and set it as default in CI instead of relying on what is the current default on the runners.

**Review by wprzytula — APPROVED** [review]

**Comment by muzarski** [comment]
> Afaik rustup is present in runner images. Maybe we should explicitly install latest stable using rustup and set it as default in CI instead of relying on what is the current default on the runners.

v2: Added a step which updates rust toolchain in CI.

**Comment by muzarski** [comment]
> > Afaik rustup is present in runner images. Maybe we should explicitly install latest stable using rustup and set it as default in CI instead of relying on what is the current default on the runners.
> 
> v2: Added a step which updates rust toolchain in CI.

@wprzytula @Lorak-mmk Do you think that we should do the same for other workflows that use rust toolchain? As of now, we only update it for `Rust / build` workflow, which does all of the clippy and format checks. However, there are other workflows that use rust compiler, e.g. `Cassandra tests / build` or `Book / build` (they do not use `clippy/rustfmt`, tho).

**Comment by wprzytula** [comment]
> Do you think that we should do the same for other workflows that use rust toolchain?

I can see two approaches. 
1. Unified CI, if we always assume that we only check on two Rust versions: latest stable and MSRV. This will hopefully result in less CI failures due to such problems as toolchain version difference locally vs in CI.
2. More than one recent stable toolchain version is used. E.g. latest and the last-before-the-latest. I think that it's a shot in a foot, because even more inconvenience will follow from such dualism.

**Comment by Lorak-mmk** [comment]
I agree, let's use latest stable everywhere, and msrv in msrv pipeline

**Comment by muzarski** [comment]
v2: using latest stable version of rust toolchain in CI (min_rust is exception - using MSRV)

**Review by wprzytula — APPROVED** [review]

**Comment by wprzytula** [comment]
@Lorak-mmk Please re-review and merge.

**Pull request merged**